core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -185,7 +180,13 @@ private static FeedItem createFeedItem(Feed feed, FastDocumentFile file, Context
                     && existingItem.getMedia().getDownload_url().equals(file.getUri().toString())
                     && file.getLength() == existingItem.getMedia().getSize()) {
                 // We found an old file that we already scanned. Re-use metadata.
+
+                // Old file's file_url can be null, which will get transferred over to a new file when updateFromOther
+                // is called. That's why we must manually set the URl afterwards.
```

Comment:
Doesn't this somehow conflict with adding the file url to `updateFromOther`?

Reply:
Yes it does and that's exactly why it's here (updateFromOther would delete the URL). It's a hack that I don't like, but I'm not sure how to solve it. updateFromOther needs to update the URL, because in other cases, new item HAS the proper URL and old item does not. Maybe we add a flag to this call instead?

Reply:
I don't really understand where this comes from. Why do we need to update the url now and didn't need it before your change? Shouldn't that be unrelated?

Reply:
Previously, file_url was null in some cases, preventing deletion from working, so I had to add it to updateFromOther. But I cannot reproduce that anymore. Will revert that code and maybe re-add it in the future if it reoccurs.